pkg/apis/vsphereprovider/v1alpha1/vsphereproviderstatus_types.go

```diff
@@ -1,33 +1,73 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// VSphereMachineProviderConditionType is a valid value for VSphereMachineProviderCondition.Type
+type VSphereMachineProviderConditionType string
+
+// Valid conditions for an vSphere machine instance
+const (
+	// MachineCreation indicates whether the machine has been created or not. If not,
+	// it should include a reason and message for the failure.
+	MachineCreation VSphereMachineProviderConditionType = "MachineCreation"
+)
+
+// VSphereMachineProviderConditionReason is reason for the condition's last transition
+type VSphereMachineProviderConditionReason string
+
+const (
+	// MachineCreationSucceeded indicates machine creation success
```

Comment:
Nit: the rest of the comments end with a period
```suggestion
	// MachineCreationSucceeded indicates machine creation success.
```